lib/lacework.bash

```diff
@@ -0,0 +1,276 @@
+    fi
+    #if LW_API_SECRET is defined locally, use that if the buildkite environment variable is absent
+    if [ -z "${API_KEY_SECRET_ENV_VAR}"  ]; then
+         export API_KEY_SECRET_ENV_VAR="${LW_API_SECRET:-}" 
+    fi
+
+    export PROFILE="${BUILDKITE_PLUGIN_LACEWORK_PROFILE:-}" 
+
+    if [ -z "${API_KEY_ENV_VAR}" ] || [ -z "${API_KEY_SECRET_ENV_VAR}" ]; then
+        if [ -z "${PROFILE}" ] && [ "${SCAN_TYPE}" != "vulnerability" ]; then
+            echo "ERROR: Missing Lacework API Key and Secret or profile" >&2
+            annotate_build_no_token
+            exit 1
+        fi
+    fi
+
+    
+
+    # need to export the below ENV variables for IAC to work since you can't pass those via CLI
+    if [ "${SCAN_TYPE}" == "iac" ]; then
+        export IAC_SCAN_TYPE="${BUILDKITE_PLUGIN_LACEWORK_IAC_SCAN_TYPE:-}"
+        export LW_ACCOUNT="${ACCOUNT_NAME}"
+        export LW_API_KEY="${API_KEY_ENV_VAR}"
+        export LW_API_SECRET="${API_KEY_SECRET_ENV_VAR}"
+
+        if [ -z "${IAC_SCAN_TYPE}" ] || [ -z "${LW_ACCOUNT}" ] ||  [ -z "${LW_API_KEY}" ] ||  [ -z "${LW_API_SECRET}" ]; then
+            if [ -z "${PROFILE}" ]; then
+                echo "ERROR: Missing config related to IAC scans. Need the following: IAC_SCAN_TYPE, LW_ACCOUNT, LW_API_KEY, LW_API_SECRET" >&2
+                exit 1
+            fi
+        fi
+    fi
+
+    #vuln scan related env variables
+    if [ "${SCAN_TYPE}" == "vulnerability" ]; then
+        export ACCESS_TOKEN_ENV_VAR="${BUILDKITE_PLUGIN_LACEWORK_ACCESS_TOKEN_ENV_VAR:-}"
+        export VULNERABILITY_SCAN_REPOSITORY="${BUILDKITE_PLUGIN_LACEWORK_VULNERABILITY_SCAN_REPOSITORY:-}"
+        export VULNERABILITY_SCAN_TAG="${BUILDKITE_PLUGIN_LACEWORK_VULNERABILITY_SCAN_TAG:-}"
+        if [ -z "${ACCESS_TOKEN_ENV_VAR}" ] || [ -z "${VULNERABILITY_SCAN_REPOSITORY}" ] || [ -z "${VULNERABILITY_SCAN_TAG}" ]; then
+            echo "ERROR: Missing config related to vulnerability scans. Need the following: ACCESS_TOKEN_ENV_VAR, VULNERABILITY_SCAN_REPOSITORY, VULNERABILITY_SCAN_TAG" >&2
+            exit 1
+        fi
+    fi
+
+    export FAIL_LEVEL="${BUILDKITE_PLUGIN_LACEWORK_FAIL_LEVEL:-}"
+
+}
+
+# Lacework Scans based on the provided scan tool
+function lacework_scan() {
+    case "${SCAN_TYPE}" in
+        sca)
+        lacework_sca
+        ;;
+
+        iac)
+        lacework_iac
+        ;;
+
+        vulnerability)
+        lacework_vulnerability
+        ;;
+
+        sast)
+        lacework_sast
+        ;;
+    esac
+}
+
+
+function lacework_sca() {
+
+    echo "--- Running Lacework SCA scan"
+
+    CMD=(
+    lacework
+    )
+
+    if [ -n "$PROFILE" ]; then
+        CMD+=(
+            --profile "$PROFILE"
+        )
+    else
+        CMD+=(
+        --account "${ACCOUNT_NAME}"
+        --api_key "${API_KEY_ENV_VAR}"
+        --api_secret "${API_KEY_SECRET_ENV_VAR}"
+    )
+    fi  
+
+    CMD+=(
+        sca scan .
+        --save-results
+    )
+        
+
+    echo "${CMD[@]}"
```

Comment:
```suggestion

```
Although the Build Logs will redact Secrets matching certain patterns, we shouldn't expose this information in first place.